### PR TITLE
Feature/cdap 14626 remove metrics migration

### PR DESCRIPTION
**Pull request by yaojiefeng**
JIRA: https://issues.cask.co/browse/CDAP-14626
Build: https://builds.cask.co/browse/CDAP-DUT6681

With 6.0, we do not need the data migration from v2 metrics to v3 metrics table any more. We can also rename the metrics table name to the correct one.

### CODE REVIEW

File: cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java [hunk 1]
```diff
@@ -651,15 +651,12 @@
...
-    // legacy name, we no longer use Kafka for metrics, we use TMS
-    public static final String KAFKA_META_TABLE = "metrics.kafka.meta.table";
 
-    // NOTE: "v2" to avoid conflict with data of older metrics system
-    public static final String DEFAULT_ENTITY_TABLE_NAME = "metrics.v2.entity";
-    public static final String DEFAULT_METRIC_TABLE_PREFIX = "metrics.v2.table";
+    // part of table name for metrics
+    public static final String KAFKA_META_TABLE = "metrics.tms.meta.table";
+    public static final String DEFAULT_ENTITY_TABLE_NAME = "metrics.entity";
+    public static final String DEFAULT_METRIC_V3_TABLE_PREFIX = "metrics.table";
```
Comment (CuriousVini) [inline cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java]: `DEFAULT_METRIC_V3_TABLE_PREFIX ` - do we still need v3?

**Review by anew — APPROVED** [review]
LGTM

File: cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java [hunk 2]
```diff
@@ -651,15 +651,12 @@
...
 
     public static final String TOPIC_PREFIX = "metrics.topic.prefix";
-    // legacy name, we no longer use Kafka for metrics, we use TMS
-    public static final String KAFKA_META_TABLE = "metrics.kafka.meta.table";
 
-    // NOTE: "v2" to avoid conflict with data of older metrics system
-    public static final String DEFAULT_ENTITY_TABLE_NAME = "metrics.v2.entity";
-    public static final String DEFAULT_METRIC_TABLE_PREFIX = "metrics.v2.table";
+    // part of table name for metrics
+    public static final String KAFKA_META_TABLE = "metrics.tms.meta.table";
```
Comment (CuriousVini) [inline cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java]: we should also rename the constant.

File: cdap-common/src/main/resources/cdap-default.xml [hunk 3]
```diff
@@ -2189,10 +2189,10 @@
   </property>
 
   <property>
-    <name>metrics.kafka.meta.table</name>
-    <value>metrics.kafka.meta</value>
+    <name>metrics.tms.meta.table</name>
+    <value>metrics.tms.meta</value>
```
Comment (CuriousVini) [inline cdap-common/src/main/resources/cdap-default.xml]: should we just call it `metrics.meta`?

File: cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java (hunk 1 above)
Comment (CuriousVini) [reply cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java]: Also remove `METRICS_DATA_MIGRATION_SKIP` from Constants?

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java [hunk 4]
```diff
@@ -115,28 +110,14 @@ protected MetricsTable getOrCreateMetricsTable(String tableName, DatasetProperti
...
-      // metrics tables are in the system namespace
-      DatasetId v2TableId = NamespaceId.SYSTEM.dataset(v2TableName);
-      MetricsTable v2Table = dsFramework.getDataset(v2TableId, ImmutableMap.of(), null);
-
       props.add(HBaseTableAdmin.PROPERTY_SPLITS,
                 GSON.toJson(getV3MetricsTableSplits(cConf.getInt(Constants.Metrics.METRICS_HBASE_TABLE_SPLITS))));
       props.add(Constants.Metrics.METRICS_HBASE_TABLE_SPLITS,
                 cConf.getInt(Constants.Metrics.METRICS_HBASE_TABLE_SPLITS));
 
       DatasetId v3TableId = NamespaceId.SYSTEM.dataset(v3TableName);
```
Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java]: rename `v3TableId` variable as well.

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java [hunk 5]
```diff
@@ -115,28 +110,14 @@ protected MetricsTable getOrCreateMetricsTable(String tableName, DatasetProperti
...
     try {
-      String v2TableName = cConf.get(Constants.Metrics.METRICS_TABLE_PREFIX,
-                                     Constants.Metrics.DEFAULT_METRIC_TABLE_PREFIX + ".ts." + resolution);
-
-      // metrics tables are in the system namespace
-      DatasetId v2TableId = NamespaceId.SYSTEM.dataset(v2TableName);
-      MetricsTable v2Table = dsFramework.getDataset(v2TableId, ImmutableMap.of(), null);
-
       props.add(HBaseTableAdmin.PROPERTY_SPLITS,
                 GSON.toJson(getV3MetricsTableSplits(cConf.getInt(Constants.Metrics.METRICS_HBASE_TABLE_SPLITS))));
```
Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java]: rename `getV3MetricsTableSplits` method name

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java [hunk 6]
```diff
@@ -115,28 +110,14 @@ protected MetricsTable getOrCreateMetricsTable(String tableName, DatasetProperti
   protected MetricsTable getOrCreateResolutionMetricsTable(String v3TableName, TableProperties.Builder props,
```
Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricDatasetFactory.java]: rename `v3TableName` method arg

File: cdap-common/src/main/resources/cdap-default.xml (hunk 3 above)
Comment (yaojiefeng) [reply cdap-common/src/main/resources/cdap-default.xml]: I will remove tms in the name

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java [hunk 7]
```diff
@@ -122,14 +111,12 @@
                                    @Assisted Set<Integer> topicNumbers,
                                    @Assisted MetricsContext metricsContext,
                                    @Assisted Integer instanceId) {
-    this(cConf, datasetFramework, metricDatasetFactory, messagingService,
-         schemaGenerator, readerFactory, metricStore, topicNumbers, metricsContext, 1000, instanceId,
-         cConf.getBoolean(Constants.MetricsProcessor.METRICS_DATA_MIGRATION_SKIP, false));
+    this(cConf, metricDatasetFactory, messagingService,
+         schemaGenerator, readerFactory, metricStore, topicNumbers, metricsContext, 1000, instanceId);
```
Comment (CuriousVini) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java:115]: Is `instanceId` used anywhere other than for metrics migration? If not should we remove that too?

Comment (yaojiefeng) [reply cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java:115]: The instance id is not required as a field in the class and I already removed it, it still needs to be passed in the constructor to construct several metric names, e.g, `processMetricName`, `metricsPrefixForDelayMetrics`

**Review by CuriousVini — APPROVED** [review]
lgtm.

**Pull request merged**